### DISCUSSION TITLE
How to set DisableUpdate to false

### DISCUSSION THREAD

**Discussion by kingbj940429** [actions/actions-runner-controller#3589] [Questions]
DisableUpdate is set to true by default, so I can't run Runner version 2.316, and manually updating to 2.317 it is very inconvenient.

I set the related environment variable in the env, but it still doesn't work as expected.

Is there any way to set DisableUpdate to false?

**Comment by nikola-jokic** [comment]
Hey :wave: 

Are you asking about the `disable update` for the scale set mode, or the legacy mode?

**Reply by kingbj940429** [reply]
@nikola-jokic 

Hey 👋

I'm using gha-runner-scale-set helm chart v0.8.1 and gha-runner-scale-set-controller helm chart v0.8.1

**Reply by nikola-jokic** [reply]
Hey, in that case, you probably wouldn't gain anything if you disable update. In most cases, your runner would receive a job and exit. The runner is ephemeral, and if it is spawned in response to a job, and not to fulfill the min runners requirement, it will likely not upgrade itself before it accepts the job.
But even if it did, it would be a bad choice since each job would wait for the runner to first upgrade and then start running it.
To always stay on the latest version, you can use the `:latest` tag for your runner image. However, I would recommend you periodically upgrade the runner to the target version. In case there is a bug in the hook, or something, your workflows would start failing, and it would be harder to diagnose the issue.

**Reply by kingbj940429** [reply]
@nikola-jokic 

the provided runner image is not ubuntu focal(it is jammy), so I create a ubuntu focal version of the runner image through a custom Dockerfile. so I cannot use the provided `:latest` tag.

It is inconvenient to have to build the image every time a new version of the runner is released.

Additionally, sometimes I do not realize that the runner has been updated, and when I run the runner, I get a message saying that the current version is deprecated.

To prevent this, I want to set `DisableUpdate: false`

Is there a better way to handle this?

<b>ps.</b>
Is manually building the image every time a new version of the runner is released the correct approach?

Anyway, manually building it is not difficult, but I have no way of knowing when a new version of the runner has been released until I see the error message at runner.

**Reply by nikola-jokic** [reply]
You can automate building images. Take a look at the workflow we have in our repo: https://github.com/actions/actions-runner-controller/blob/master/.github/workflows/arc-update-runners-scheduled.yaml
We periodically check on a cron job if a new runner has been released, and we create a PR. Then we merge that PR that triggers runner build. You can do something similar :relaxed: 

I still don't think that disable update would be the correct approach. Imagine for each job your runner needs to download the latest version and start it.

**Reply by kingbj940429** [reply]
I 100% understood.

I will do like you say.

but still I wonder do you have a plan to  provide an option to set `DisableUpdate: false` at gha-runner-scale-set helm chart v0.8.1 and gha-runner-scale-set-controller helm chart v0.8.1 ?

thank you for your commend :)
